### PR TITLE
support configuration of kube-proxy IPVS tcp,tcpfin,udp timeout

### LINKED ISSUE #84041: Kube-proxy should support setting ipvs tcp/tcpfin/udp timeout when proxy-mode is ipvs
**What would you like to be added**:
kube-proxy can add three new options to set the ipvs tcp/tcpfin/udp timeout when proxy-mode is ipvs, such as:
``` shell
# kube-proxy new options
--ipvs-tcp-timeout duration
--ipvs-tcpfin-timeout duration
--ipvs-udp-timeout duration

# corresponding to the settings below
root@my-master01:~# ipvsadm -ln --timeout
Timeout (tcp tcpfin udp): 900 120 300
```

**Why is this needed**:
kube-proxy has some options to set kernel parameters related to netfilter conntrack, such as:
`net.netfilter.nf_conntrack_tcp_timeout_established` can be set by `--conntrack-tcp-timeout-established` 
`net.netfilter.nf_conntrack_tcp_timeout_close_wait` can be set by `--conntrack-tcp-timeout-close-wait`

So, when kube-proxy's proxy-mode is ipvs, we also want to set the ipvs tcp/tcpfin/udp timeout by kube-proxy. 

currently, we can only set it through the command, like `ipvsadm --set 900 120 60`, but when k8s node is restarted, these changes will be invalid(of course, use rc.local or systemd is also a way)

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR allows IPVS tcp, tcpfin and udp timeouts to be configurable via kube-proxy flags/config. 

The following flags are being added:
```
--ipvs-tcp-timeout <duration>
--ipvs-tcpfin-timeout <duration>
--ipvs-udp-timeout <duration>
```

The flags above are equivalent to running the following `ipvsadm` command on any linux node:
```
ipvsadm --set <tcp-timeout> <tcpfin-timeout> <udp-timeout>
```

`ipvsadm --set` according to the [ipvsadm man page](https://linux.die.net/man/8/ipvsadm):
```
--set tcp tcpfin udp
    Change the timeout values used for IPVS connections. This command always takes 3 parameters, representing the timeout values (in seconds) for TCP sessions, TCP sessions after receiving a FIN packet, and UDP packets, respectively. A timeout value 0 means that the current timeout value of the corresponding entry is preserved. 
```
The default values for the new kube-proxy flags are 0 because passing in 0 to `ipvsadm --set` means the current timeout value on the system is preserved. 

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/84041

Some other issues where users have requested the need for configuring the IPVS timeouts:
https://github.com/kubernetes/kubernetes/issues/81775
https://github.com/kubernetes/kubernetes/issues/81308
https://github.com/kubernetes/kubernetes/issues/77903
https://github.com/kubernetes/kubernetes/issues/71358

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
add kube-proxy flags --ipvs-tcp-timeout, --ipvs-tcpfin-timeout, --ipvs-udp-timeout to configure IPVS connection timeouts.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by andrewsykim** [comment]
/assign @lbernail @thockin 

Do we need a KEP for this? I think the use-case is pretty clear but maybe I missed something.

**Comment by lbernail** [comment]
/lgtm

I wonder if we should set defaults that would be better for most use cases
Linux defaults are:
- tcp: 900s
- tcpfin: 120s
- udp: 300s

However the standard conntrack (used by iptables) defaults to:
- nf_conntrack_tcp_timeout_established: 86400s
- nf_conntrack_tcp_timeout_fin_wait: 120s
- udp: 30s (for nf_conntrack_udp_timeout, for use cases with "established" connections it's nf_conntrack_udp_timeout_stream: 180 but I really think 90+% of UDP issues we have are dns related so 30s should be a lot better)

So I think the tcp defaults make sense in most cases but we could lower udp a lot. For the tcp use-case, the most interesting change would actually be to lower tcp_retries (which would probably help with iptables too)

**Comment by andrewsykim** [comment]
I was hesitate to change defaults here since it could unexpectedly override user configured timeouts on the system (set using ipvsadm --set). Maybe it's fine though if most users don't change these timeouts anyways. Need to think about this more.

**Comment by lbernail** [comment]
@andrewsykim we can always change the default timeout later
Let's get this merged it will help many users
/approve

**Review by lbernail — APPROVED** [review]

File: pkg/proxy/apis/config/types.go [hunk 1]
```diff
@@ -58,6 +58,15 @@ type KubeProxyIPVSConfiguration struct {
 	// strict ARP configure arp_ignore and arp_announce to avoid answering ARP queries
 	// from kube-ipvs0 interface
 	StrictARP bool
+	// tcpTimeout is the timeout value used for idle IPVS TCP sessions.
```
Comment (thockin) [inline pkg/proxy/apis/config/types.go]: optional: would it make sense to group these in a `timeout` struct ?

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by liggitt** [comment]
/test pull-kubernetes-bazel-test

Unit tests added since this was opened require update of test fixtures

**Pull request merged**